pkg/cmd/opts/version.go

```diff
@@ -3,16 +3,96 @@ package opts
 import (
 	"strings"
 
-	"github.com/jenkins-x/jx/pkg/versionstream/versionstreamrepo"
-
-	"github.com/jenkins-x/jx/pkg/versionstream"
-
+	"github.com/jenkins-x/jx/pkg/config"
+	"github.com/jenkins-x/jx/pkg/envctx"
+	"github.com/jenkins-x/jx/pkg/kube"
 	"github.com/jenkins-x/jx/pkg/log"
 	"github.com/jenkins-x/jx/pkg/table"
 	"github.com/jenkins-x/jx/pkg/util"
 	"github.com/jenkins-x/jx/pkg/version"
+	"github.com/jenkins-x/jx/pkg/versionstream"
+	"github.com/jenkins-x/jx/pkg/versionstream/versionstreamrepo"
 )
 
+// EnvironmentContext gets or creates a team context with the common values for working with requirements, team settings
+// and version resolvers.
+//
+// if preferRequirementsFile is true we look for the local `jx-requirements.yml` file first and then fall back to the
+// development environment settings; otherwise we try to use the requirements from the environment if present
+func (o *CommonOptions) EnvironmentContext(dir string, preferRequirementsFile bool) (*envctx.EnvironmentContext, error) {
+	if o.envctx != nil {
+		return o.envctx, nil
+	}
+	var err error
+	tc := &envctx.EnvironmentContext{}
+	tc.GitOps, tc.DevEnv = o.GetDevEnv()
+	if tc.DevEnv == nil {
+		tc.DevEnv = kube.CreateDefaultDevEnvironment("jx")
+	}
+	teamSettings := tc.TeamSettings()
+
+	// lets default to local file system for the requirements as we are often invoked before we've created the cluster
```

Comment:
Would it not be better to default to the connected cluster. Otherwise I can envisage a situation where I execute the command inside a checkout of a enviroment but connected to a different cluster, and getting odd results.

Reply:
the main reason for this logic is that sometimes we are invoking this `EnvironmentContext` before we have even created any `Environment` resources (e.g. early `jx` commands in the boot pipeline) so there's nothing to connect to really.

I've modified the comment to make this all a bit more clear: https://github.com/jenkins-x/jx/blob/00c0cee8775cc8d8ed2fab0ba158a8997330a9f2/pkg/cmd/opts/version.go#L36

Reply:
I wasn't questioning why you would look on the local filesystem, just why it was the default. 

For the case that we are early in the boot pipeline it's obvious it needs to look at the file system. In this case if we defaulted to the connected cluster it wouldn't find the resources there so would continue to look at the file system.

However after boot it would then default to the connected cluster.

Reply:
got it - yeah I've added a boolean flag now to indicate which mode to use so its more clear